rust/tw_bitcoin/src/lib.rs

```diff
@@ -0,0 +1,402 @@
+                            index,
+                            &bitcoin::sighash::Prevouts::All(&prevouts),
+                            TapSighashType::Default,
+                        )
+                        .map_err(|_| Error::Todo)?;
+
+                    let message = secp256k1::Message::from_slice(hash.as_ref()).unwrap();
+                    let updated = signer(input, message)?;
+
+                    claims.push((index, updated));
+                },
+                TxInput::P2TRScriptPath(p) => {
+                    let leaf_hash = TapLeafHash::from_script(&p.script, LeafVersion::TapScript);
+
+                    let hash = cache
+                        .taproot_script_spend_signature_hash(
+                            index,
+                            &bitcoin::sighash::Prevouts::All(&prevouts),
+                            leaf_hash,
+                            TapSighashType::Default,
+                        )
+                        .map_err(|_| Error::Todo)?;
+
+                    let message = secp256k1::Message::from_slice(hash.as_ref()).unwrap();
+                    let updated = signer(input, message)?;
+
+                    claims.push((index, updated));
+                },
+                // Skip.
+                TxInput::NonStandard { ctx: _ } => continue,
+            };
+        }
+
+        let mut tx = cache.into_transaction();
+
+        // Update the transaction with the updated scriptSig/Witness.
+        for (index, claim_loc) in claims {
+            match claim_loc {
+                ClaimLocation::Script(script) => {
+                    tx.input[index].script_sig = script;
+                },
+                ClaimLocation::Witness(witness) => {
+                    tx.input[index].witness = witness;
+                },
+            }
+        }
+
+        Ok(TransactionSigned { tx })
+    }
+}
+
+pub struct TransactionSigned {
+    tx: Transaction,
+}
+
+impl TransactionSigned {
+    pub fn serialize(&self) -> Result<Vec<u8>> {
+        let mut buffer = vec![];
+        self.tx
+            .consensus_encode(&mut buffer)
+            .map_err(|_| Error::Todo)?;
+
+        Ok(buffer)
+    }
+}
+
+pub struct TxInputsOuputs {
+    pub inputs: Vec<TxInput>,
+    pub outputs: Vec<TxOutput>,
+}
+
+#[derive(Debug, Clone)]
+pub struct InputContext {
+    pub previous_output: OutPoint,
+    pub value: Option<u64>,
+    // The condition for claiming the output.
+    pub script_pubkey: ScriptBuf,
+    // TODO: Document this.
+    pub sequence: Sequence,
+    // Witness data for Segwit/Taproot transactions.
+    // TODO: Remove this?
+    pub witness: Witness,
+}
+
+impl InputContext {
+    pub fn new(utxo: TxOut, point: OutPoint) -> Self {
+        InputContext {
+            previous_output: point,
+            value: Some(utxo.value),
+            script_pubkey: utxo.script_pubkey,
+            // Default value of `0xFFFFFFFF = 4294967295`.
+            sequence: Sequence::default(),
```

Comment:
Sequence can be different from 0 in some cases.
I guess we need to allow to set the sequence number for an input